iceoryx_hoofs/source/relocatable_pointer/base_relative_pointer.cpp

```diff
@@ -21,13 +21,17 @@ namespace iox
 {
 namespace rp
 {
-BaseRelativePointer::BaseRelativePointer(ptr_t ptr, id_t id) noexcept
+// NOLINTJUSTIFICATION NewType is trivially copyable, hence pass by value is preferred
```

Comment:
It is also small if the wrapped type is only an integer of some kind. Unless the compiler creates more hidden data than I would expect.

Trivially copyable should not be the reason, size of the object and object semantics are (some types are naturally value types). There can be large trivially copyable objects which should not be passed by value.

Please change the justification.

Reply:
True indeed, adapted.